libs/api/src/main/java/de/cyface/api/Authorizer.java

```diff
@@ -64,20 +74,32 @@ public abstract class Authorizer implements Handler<RoutingContext> {
      * The database to check which users a {@code manager} user has access to.
      */
     private final MongoClient mongoUserDatabase;
+    /**
+     * {@code true} if the {@code #request} needs to be paused when waiting for an async result.
+     * <p>
+     * This is necessary when the request body is parsed after this handler or else `403: request body already read` is
+     * thrown [DAT-749].
+     */
+    private final boolean pauseAndResume;
 
     /**
      * Creates a new completely initialized instance of this class with access to all required authentication
      * information to authorize a request and fetch the correct data.
      *
      * @param authProvider An auth provider used by this server to authenticate against the Mongo user database
      * @param mongoUserDatabase The Mongo user database containing all information about users
+     * @param pauseAndResume {@code true} if the {@code #request} needs to be paused when waiting for an async result.
+     *            This is necessary when the request body is parsed after this handler or else `403: request body
+     *            already read` if thrown [DAT-749].
      */
-    public Authorizer(final MongoAuthentication authProvider, final MongoClient mongoUserDatabase) {
+    public Authorizer(final MongoAuthentication authProvider, final MongoClient mongoUserDatabase,
+            final boolean pauseAndResume) {
```

Comment:
We had this `pauseAndResume` option before is the `Collector.RequestHandler` or `Collector.MeasurementHandler`.

This moved here to make sure all APIs use the same `Authorization` code.

I could also replace this flag with a Strategy, with a constructor which uses the default strategy (i.e. no "pause and resume before async requests" and another strategy which currently only the `Collector.MeasurementHandler` uses.)

Reply:
In this class I guess I would then have e.g.:

```
strategy.pause()
final var authentication = authProvider.authenticate(principal);
    authentication.onComplete(r -> {
        try {
           strategy.resume();
...
```

**Instead of ...**

```
 // Before async operations, pause request body parsing to not lose the body or protocol upgrades.
            // `pauseAndResume` is set to `true` when the `BodyHandler` is not executed before this handler [DAT-749]
            final var parseEnded = request.isEnded();
            if (pauseAndResume) {
                if (!parseEnded) {
                    request.pause();
                }
            }
            final var authentication = authProvider.authenticate(principal);
            authentication.onComplete(r -> {
                try {
                    // As in `SessionHandlerImpl.handle(RoutingContext)`
                    final var upgraded = request.headers().contains(HttpHeaders.UPGRADE, HttpHeaders.WEBSOCKET, true);
                    if (pauseAndResume && !parseEnded && !upgraded) {
                        request.resume();
                    }
```

Reply:
Strategy always sounds better than a flag. ;)